assets/src/components/mapPage.tsx

```diff
@@ -65,6 +65,20 @@ const MapPage = (): ReactElement<HTMLDivElement> => {
   const selectedVehicleShapes = useTripShape(selectedVehicle?.tripId || null)
   const [showVehicleCard, setShowVehicleCard] = useState<boolean>(false)
 
+  useEffect(() => {
+    if (selectedVehicle) {
+      const updatedSelectedVehicle = vehicles?.find(
+        (v) => v.id === selectedVehicle.id
+      )
+      if (updatedSelectedVehicle && isVehicle(updatedSelectedVehicle)) {
+        setSelectedVehicle(updatedSelectedVehicle)
+      } else {
+        setSelectedVehicle(null)
+        setShowVehicleCard(false)
+      }
+    }
+  }, [vehicles])
```

Comment:
**suggestion:** I think it should be possible, and preferable, to disentangle the various pieces of state here such that we don't need a `useEffect` hook to make this update, since it's all within React state and / or props. That said, I haven't fully thought through exactly how to do that. Other hooks like `useMemo` may help with preventing repetition of the computational part when `vehicles` doesn't change.

Reply:
That's fair, I'm going to re-mark this as a draft for now and do some more thinking about that after addressing the requested changes in the base branch.

Reply:
This hook has been removed in [381021d](https://github.com/mbta/skate/pull/1809/commits/381021df8e2ee873f098978875f6212cb4e9c9d3). This also modifies the state to only store the selectedVehicleId rather than the whole vehicle - that helped me as I was thinking about it to distinguish between what is static and what is live. 

I started to go down a separate rabbit hole of considering whether or not it makes sense for the map page to be preserving its own notion of a separate vehicle rather than using the `separateVehicleOrGhost` from state, but I think that consideration can be addressed outside of this PR.